src/tests/java/org/broadinstitute/dropseqrna/utils/SplitBamByCellTest.java

```diff
@@ -0,0 +1,120 @@
+
+        bamSplitter.INPUT=Arrays.asList(TEST_BAM);
+        bamSplitter.OUTPUT=outputBAM;
+        bamSplitter.OUTPUT_LIST=outputBAMList;
+        bamSplitter.NUM_OUTPUTS = 3;
+        bamSplitter.REPORT = report;
+		Assert.assertEquals(bamSplitter.doWork(), 0);
+
+        List<String> splitBAMFileList;
+		try {
+            splitBAMFileList = FileUtils.readLines(outputBAMList);
+            for (String filePath : splitBAMFileList) {
+                new File(filePath).deleteOnExit();
+            }
+
+            // Merge the split BAM files
+            final MergeSamFiles fileMerger = new MergeSamFiles();
+            List<String> args = new ArrayList<>();
+            for (String splitBAMFilePath : splitBAMFileList)  {
+                args.add("INPUT=" + splitBAMFilePath);
+            }
+            args.add("OUTPUT=" + mergedOutputBAM.getAbsolutePath());
+            Assert.assertEquals(fileMerger.instanceMain(args.toArray(new String[args.size()])), 0);
+
+            // Metrics for the input test BAM file
+            args = new ArrayList<String>();
+            args.add("INPUT=" + TEST_BAM.getAbsolutePath());
+            args.add("OUTPUT=" + originalMetrics.getAbsolutePath());
+            final CollectAlignmentSummaryMetrics originalMetricsCollector = new CollectAlignmentSummaryMetrics();
+            Assert.assertEquals(originalMetricsCollector.instanceMain(args.toArray(new String[args.size()])), 0);
+
+            // Metrics for the split and merged test BAM file
+            args = new ArrayList<String>();
+            args.add("INPUT=" + mergedOutputBAM.getAbsolutePath());
+            args.add("OUTPUT=" + mergedMetrics.getAbsolutePath());
+            final CollectAlignmentSummaryMetrics mergedMetricsCollector = new CollectAlignmentSummaryMetrics();
+            Assert.assertEquals(mergedMetricsCollector.instanceMain(args.toArray(new String[args.size()])), 0);
+
+            // Make sure the input test BAM and the merged BAM files' metrics are identical
+            TestUtils.testMetricsFilesEqual(originalMetrics, mergedMetrics);
+
+            // Compare the XC and XM tags for the input test BAM and the merged BAM files
+            final CompareBAMTagValues tagsComparator = new CompareBAMTagValues();
+            tagsComparator.INPUT_1 = TEST_BAM;
+            tagsComparator.INPUT_2 = mergedOutputBAM;
+            tagsComparator.TAGS = new ArrayList<>(Arrays.asList("XC", "XM"));
+            Assert.assertEquals(tagsComparator.doWork(), 0);
+
+			boolean t1 = FileUtils.contentEquals(report, EXPECTED_REPORT);
+			Assert.assertTrue(t1);
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+    }
+
+    private File getTempReportFile (final String prefix, final String suffix) {
+        File tempFile=null;
+
+        try {
+            tempFile = File.createTempFile(prefix, suffix);
+            tempFile.deleteOnExit();
+        } catch (IOException e1) {
+            e1.printStackTrace();
```

Comment:
I don't think you should not propagate the exception here.  Either add 'throws' to the method declaration, or wrap this in a RuntimeException so that it causes the test to fail.